programs/protocol_event/src/state/event.rs

```diff
@@ -0,0 +1,85 @@
+use anchor_lang::prelude::*;
+use crate::state::type_size::{BOOL_SIZE, CHAR_SIZE, DISCRIMINATOR_SIZE, I64_SIZE, option_size, PUB_KEY_SIZE, vec_size};
+
+#[account]
+pub struct Event {
+    pub authority: Pubkey,
+    pub payer: Pubkey,
+
+    pub category: Category,
+    pub event_group: EventGroup,
+
+    pub active: bool,
+
+    pub slug: String, // event identifier e.g. LAFCvLAG@2021-08-28
+    pub name: String, // for display purposes e.g. Los Angeles Football Club vs. LA Galaxy
+
+    pub participants: Vec<Participant>,
+
+    pub expected_start_timestamp: i64,
+    pub actual_start_timestamp: Option<i64>,
+    pub actual_end_timestamp: Option<i64>,
+}
+
+impl Event {
+    const MAX_EVENT_SLUG_LENGTH: usize = 25;
+    const MAX_EVENT_NAME_LENGTH: usize = 50;
+    const MAX_PARTICIPANTS: usize = 64;
```

Comment:
Need to think about sizing this, for most markets 64 participants will be fine - there are 48 teams in the next world cup. However, there are 128 participants in Wimbledon events if we were doing an outright; https://en.wikipedia.org/wiki/Wimbledon_Championships#Events

But we actually hit the account size limit with this

Reply:
Is this error during account initialisation? There is an initial account limit, but they can be further grown with later additional reallocation instructions.

Reply:
Looks like this is a PDA size limit of 10KiBs. Too much data!

Reply:
We had a good chat about this and I'm going to write up a new proposal for storing event participants, I will update this PR with a placeholder field for now